packages/frontend-shared/src/store/login-connect-store.ts

```diff
@@ -0,0 +1,134 @@
+import type { BannersState } from '@packages/types'
+import { defineStore } from 'pinia'
+
+interface LoginUserData {
+  fullName: string | null
+  email: string | null
+}
+
+export interface LoginConnectState {
+  isLoginConnectOpen: boolean
+  utmMedium: string
+  isLoggedIn: boolean
+  isProjectConnected: boolean
+  isConfigLoaded: boolean
+  isOrganizationLoaded: boolean
+  isMemberOfOrganization: boolean
+  hasNoRecordedRuns: boolean
+  error: boolean
+  hasRecordedRuns: boolean
+  loginError: boolean
+  userData?: LoginUserData
+  firstOpened?: number
+  latestBannerShownTime?: number
+  hasNonExampleSpec: boolean
+  promptsShown: {
+    ci1?: number
+    loginModalRecord?: number
+  }
+  bannersState: BannersState
+}
+
+// The user can be in only one status at a time.
+// These are specifically related to the dashboard
+// and the progress from logging in to recording a run.
+export const userStatuses = [
+  'isLoggedOut',
+  'needsOrgConnect',
+  'needsProjectConnect',
+  'needsRecordedRun',
+  'allTasksCompleted',
+] as const
+
+export type UserStatus = typeof userStatuses[number]
+
+export type LoginConnectBooleanField = 'isLoggedIn' | 'isProjectConnected' | 'isConfigLoaded' | 'isOrganizationLoaded' | 'isMemberOfOrganization' | 'hasRecordedRuns' | 'hasNoRecordedRuns' | 'hasNonExampleSpec'
+
+export const useLoginConnectStore = defineStore({
+  id: 'loginConnect',
+
+  state (): LoginConnectState {
+    return {
+      isLoginConnectOpen: false,
+      utmMedium: '',
+      isLoggedIn: false,
+      isProjectConnected: false,
+      isConfigLoaded: false,
+      isOrganizationLoaded: false,
+      isMemberOfOrganization: false,
+      hasNoRecordedRuns: false,
+      error: false,
+      hasRecordedRuns: false,
+      loginError: false,
+      userData: undefined,
+      hasNonExampleSpec: true, // TODO: in #23762 initialize as false and set the real value
+      promptsShown: {},
+      firstOpened: undefined,
+      bannersState: {},
+      latestBannerShownTime: undefined,
+    }
+  },
+  actions: {
+    openLoginConnectModal ({ utmMedium }: { utmMedium: string }) {
+      this.isLoginConnectOpen = true
+      this.utmMedium = utmMedium
+    },
+    closeLoginConnectModal () {
+      this.isLoginConnectOpen = false
+      this.utmMedium = ''
```

Comment:
If we do pull the login modal stuff out, then these actions would go with it along with the `setLoginError` below.